minpy/array.py

```diff
@@ -365,20 +369,41 @@ def to_array_type(arr):
     def __str__(self):
         return str(self.get_data(ArrayType.NUMPY))
 
+    def __repr__(self):
+        return self.__str__()
+
     @property
     def node(self):
         """ get node which contains derivative information from this array """
         return self._node
 
     def has_type(self, atype):
-        """Return whether array data of given type exists in the underlying storage.
+        """ Return whether array data of given type exists in the underlying storage.
         """
         return atype in self._data.keys()
 
-    def reshape(self, new_shape):
-        """ Function for reshape this array """
+    def reshape(self, *args):
+        """ Function for reshape this array
+
+        Usage example:
+        Assume a = np.ones([10, 10])
+        b = a.reshape([5, 20])
+        b = a.reshape(5, 20)
+        :param args: a single iterable or a sequence of ints representing a new shape
+        :return: reshaped array (minpy array)
+        """
+        # Although this usage is not documented in numpy official doc, it is renowned and
+        # widely used in practice
+        if len(args) == 1 and isinstance(args[0], collections.Iterable):
```

Comment:
Be careful about the `order` argument. I suggest add the `order` as keyword argument but check it to be 'C', throw error if not.